modules/@apostrophecms/schema/ui/apos/components/AposInputArray.vue

```diff
@@ -244,6 +246,50 @@ export default {
       if (this.field.max && value.length > this.field.max) {
         return 'max';
       }
+      if (value.length && this.field.fields && this.field.fields.add) {
+        const [ uniqueFieldName, uniqueFieldSchema ] = Object.entries(this.field.fields.add).find(([ , subfield ]) => subfield.unique);
+        if (uniqueFieldName) {
+          const duplicates = this.next
+            .map(item => Array.isArray(item[uniqueFieldName])
+              ? item[uniqueFieldName].map(i => i._id).sort().join('|')
+              : item[uniqueFieldName])
+            .filter((item, index, array) => array.indexOf(item) !== index);
+
+          if (duplicates.length) {
+            duplicates.forEach(duplicate => {
+              if (uniqueFieldSchema.type === 'relationship') {
+                nextTick(() => {
+                  duplicate.split('|').forEach(item => {
+                    this.$el.querySelectorAll(`[data-id="${item}"]`).forEach(relation => {
```

Comment:
This should be happening properly through Vue. We have existing capabilities to indicate the state and errors of an input field via what is provided with v-model etc. Direct DOM manipulation for this feels like a hint that something is wrong. It breaks encapsulation and could damage our ability to override the components with different visual presentation.